src/PIL/SgiImagePlugin.py

```diff
@@ -156,24 +149,15 @@ def _save(im: Image.Image, fp: IO[bytes], filename: str | bytes) -> None:
     # Run-Length Encoding Compression - Unsupported at this time
     rle = 0
 
-    # Number of dimensions (x,y,z)
-    dim = 3
     # X Dimension = width / Y Dimension = height
     x, y = im.size
-    if im.mode == "L" and y == 1:
-        dim = 1
-    elif im.mode == "L":
-        dim = 2
     # Z Dimension: Number of channels
     z = len(im.mode)
-
-    if dim in {1, 2}:
-        z = 1
-
-    # assert we've got the right number of bands.
-    if len(im.getbands()) != z:
-        msg = f"incorrect number of bands in SGI write: {z} vs {len(im.getbands())}"
-        raise ValueError(msg)
+    # Number of dimensions (x,y,z)
+    if im.mode == "L":
+        dim = 1 if y == 1 else 2
+    else:
+        dim = 3
```

Comment:
Shall we use more descriptive name? (and below)
```suggestion
        dimensions = 1 if y == 1 else 2
    else:
        dimensions = 3
```

Reply:
I've pushed a commit with `dimension`. It matches
https://github.com/python-pillow/Pillow/blob/1788ab7887beb80ee89cca9b853a391d7f026a4c/src/PIL/SgiImagePlugin.py#L73-L74

and the NAME in 'The Header' at https://paulbourke.net/dataformats/sgirgb/sgiversion.html